cogs/info.py

```diff
@@ -0,0 +1,72 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) idoneam (2016-2019)
+#
+# This file is part of Canary
+#
+# Canary is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Canary is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Canary. If not, see <https://www.gnu.org/licenses/>.
+
+# discord-py requirements
+import discord
+from discord.ext import commands
+import asyncio
+
+from .utils.paginator import Pages
+import math
+
+
+class Info(commands.Cog):
+    def __init__(self, bot):
+        self.bot = bot
+
+    @commands.command()
+    async def roles(self, ctx):
+        """Returns list of all roles in server"""
+        roleNames = list(map(lambda role: role.name + "\n", ctx.guild.roles))
+        roleNames.reverse()
+        p = Pages(ctx,
+                  item_list=roleNames,
+                  title="All roles in server",
+                  display_option=(3, 20),
+                  editable_content=False)
+
+        await p.paginate()
+
+    @commands.command()
+    async def inrole(self, ctx, *, r):
+        """Returns list of users in the specified role"""
+        members = []
+        for role in ctx.guild.roles:
+            if role.name.lower() == r.lower():
+                members = role.members
+                break
+
+        if (len(members) == 0): return
```

Comment:
There are some roles that exist but have 0 members, which are different from roles that don't exist – it might be good to display something for empty (but existing) roles

Reply:
Suggestion: add a flag `found` and return `if not found:`, or start with `members = None` and return `if members is None:`